microservice/challenge-1/app42/backend/server.ts

```diff
@@ -0,0 +1,43 @@
+import http = require("http");
+import log4js = require("log4js");
+
+import appManager = require("./src/AppManager");
+import config = require("./src/config");
+import db from "./src/database/DbConnection";
+
+const app = appManager.createApp();
+
+const logger = log4js.getLogger(module.filename);
+logger.level = "debug";
+
+app.set("port", config.server.port);
+const server = http.createServer(app);
+
+const connect = async () => {
+    let retries = 5;
+    while (retries > 0) {
+        try {
+            await db.sync({force: true});
+            logger.info("[i] database connected");
+            server.listen(config.server.port);
+            server.on("error", (error) => {
```

Comment:
For each retry, the `anonymous` methods used with events might be redeclared, what would be the best solution?

Reply:
That's a silly mistake. So much better to pull out the server.on("Error") after defining server on Linde 14.